bridge/contracts/HecBridgeSplitter.sol

```diff
@@ -95,81 +77,147 @@ contract HecBridgeSplitter is OwnableUpgradeable, PausableUpgradeable {
+	/// @param sendingAsset Asset that is used for bridge
 	/// @param sendingAssetInfos Array Data used purely for sending assets
-	/// @param fees Amounts of native coin amounts for bridge
-	/// @param callDatas CallDatas from lifi sdk
-	/// @param useSquid use Squid or Lifi
-	/// @param squidTargetAddress use in executing squid bridge contract
-	function Bridge(
-		SendingAssetInfo[] memory sendingAssetInfos,
-		uint256[] memory fees,
-		bytes[] memory callDatas,
-		bool useSquid,
-		address squidTargetAddress
+	/// @param callTargetAddress use in executing squid bridge contract
+	function bridge(
+		address sendingAsset,
+		SendingAssetInfo[] calldata sendingAssetInfos,
+		address callTargetAddress
 	) external payable {
 		require(
 			sendingAssetInfos.length > 0 &&
 				sendingAssetInfos.length <= CountDest &&
-				sendingAssetInfos.length == callDatas.length,
-			'Splitter: bridge or swap call data is invalid'
+				isInWhiteList(callTargetAddress),
+			'Bridge: Invalid parameters'
 		);
-		require(useSquid && squidTargetAddress != address(0), 'Splitter: squid router is invalid');
-
-		address callTargetAddress = useSquid ? squidTargetAddress : LiFiBridge;
-		for (uint256 i = 0; i < sendingAssetInfos.length; i++) {
-			if (sendingAssetInfos[i].sendingAssetId != address(0)) {
-				IERC20Upgradeable srcToken = IERC20Upgradeable(sendingAssetInfos[i].sendingAssetId);
-
-				require(
-					srcToken.allowance(msg.sender, address(this)) > 0,
-					'ERC20: transfer amount exceeds allowance'
-				);
 
-				srcToken.safeTransferFrom(msg.sender, address(this), sendingAssetInfos[i].sendingAmount);
-				srcToken.approve(callTargetAddress, sendingAssetInfos[i].sendingAmount);
-			}
-
-			if (msg.value > 0 && fees.length > 0 && fees[i] > 0) {
-				(bool success, ) = payable(callTargetAddress).call{value: fees[i]}(callDatas[i]);
-				require(success, 'Splitter: bridge swap transaction was failed');
-				emit CallData(success, callDatas[i]);
+		// Receive asset
+		_receiveAssets(sendingAsset, sendingAssetInfos, callTargetAddress);
+
+		uint length = sendingAssetInfos.length;
+		for (uint i = 0; i < length; i++) {
+			bytes memory callData = sendingAssetInfos[i].callData;
+			uint256 fee = sendingAssetInfos[i].bridgeFee;
+			if (fee > 0) {
+				(bool success, bytes memory result) = payable(callTargetAddress).call{value: fee}(callData);
+				if (!success) revert(_getRevertMsg(result));
+				emit MakeCallData(success, callData, msg.sender);
 			} else {
-				(bool success, ) = payable(callTargetAddress).call(callDatas[i]);
-				require(success, 'Splitter: bridge swap transaction was failed');
-				emit CallData(success, callDatas[i]);
+				(bool success, bytes memory result) = payable(callTargetAddress).call(callData);
+				if (!success) revert(_getRevertMsg(result));
+				emit MakeCallData(success, callData, msg.sender);
 			}
 		}
-
 		emit HectorBridge(msg.sender, sendingAssetInfos);
 	}
 
+	// Receive asset
```

Comment:
@xiangdev1016 we may want to add a withdraw func for admin to withdraw tokens which may get stuck in this bridge contract when the bridge fails for some unexpected reason. You can use something like this https://github.com/Hector-Network/hector-contracts/blob/befa2e472db7462bf29e4aa92dd880cb8b9353c5/pay/contracts/subscription/HectorSubscriptionTreasury.sol#L62